src/app/pages/send/send-crypto-asset-form/family/bitcoin/hooks/use-validate-total-spend.ts

```diff
@@ -0,0 +1,35 @@
+import { useState } from 'react';
+
+import { Money, createMoney } from '@shared/models/money.model';
+
+import { sumMoney } from '@app/common/money/calculate-money';
+import { useCurrentNativeSegwitAddressBalance } from '@app/query/bitcoin/balance/bitcoin-balances.query';
+
+export function useValidateTotalSpend(amount?: Money) {
+  const [showInsufficientBalanceError, setShowInsufficientBalanceError] = useState(false);
+  const balance = useCurrentNativeSegwitAddressBalance();
+
+  return {
+    showInsufficientBalanceError,
+    onValidateTotalSpend(feeValue: number) {
+      const feeAsMoney = createMoney(feeValue, 'BTC');
+
+      if (!amount || amount.symbol !== 'BTC') {
+        if (feeAsMoney.amount.isGreaterThan(balance.amount)) {
+          setShowInsufficientBalanceError(true);
+          return;
+        }
+      }
+
+      // Check amount + fee only for BTC
+      if (amount?.symbol === 'BTC') {
+        const totalSpend = sumMoney([amount, feeAsMoney]);
+
+        if (totalSpend.amount.isGreaterThan(balance.amount)) {
+          setShowInsufficientBalanceError(true);
+          return;
+        }
+      }
+    },
+  };
+}
```

Comment:
I wonder if this should be two diff hooks? Also name is not so descriptive validate total spend of what?

Reply:
useValidateBtcTotalSend?
wonder how to better name second hook?

Reply:
Is the other instance just for Brc20? If so then yeah

`useValidateTotalBrcSpend`
`useValidateTotalBrc20Spend`

?

Reply:
we also use this validation when send inscriptions (to validate only fees)

Reply:
The challenge is how we can best share this logic in a way that does involve a fn that does many things, breaking [this rule](https://github.com/ryanmcdermott/clean-code-javascript#dont-use-flags-as-function-parameters).